src/otg/datasource/gwas_catalog/study_index.py

```diff
@@ -243,7 +243,7 @@ def from_source(
         catalog_studies: DataFrame,
         ancestry_file: DataFrame,
         sumstats_lut: DataFrame,
-    ) -> StudyIndex:
+    ) -> GWASCatalogStudyIndex:
```

Comment:
Great catch! objects of class `GWASCatalogStudyIndex` are also instances of the `StudyIndex` as well. Is this something mypy enforces or you have just seen it?

Reply:
the problem starts when you start using the resulting object. Methods for the child class are not "typed" to be available if you return the parent type. In these cases the more specific the better.